src/main/resources/xlr_xldeploy/XLDVersionsTile.py

```diff
@@ -11,14 +11,29 @@
 
 from xlr_xldeploy.XLDeployClientUtil import XLDeployClientUtil
 
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+
 if not xldeployServer:
     raise Exception("XL Deploy server ID must be provided")
 
+#check the range of values for numberOfDays,
+if not 1 <= numberOfDays <= maxDays:
+    numberOfDays = maxDays
```

Comment:
Can we log something here? So that it's clear in the logs that the data was truncated?